android/app/src/main/java/network/perun/app/MainActivity.java

```diff
@@ -0,0 +1,170 @@
+            // Create the initial balances of the channel, we start with 2000 and bob with 1000.
+            BigInts initBals = Prnm.newBalances(new BigInt(2000), new BigInt(1000));
+            // Send the proposal. Once the channel was accepted, the node will handle the
+            // channel updates and onChain watching.
+            node.propose(bob, initBals, "10.0.2.2", 5750);
+        } catch (Exception e) {
+            Log.e("prnm", e.toString());
+        }
+    }
+}
+
+class Node implements ChannelAcceptor {
+    Client client;
+
+    public Node(Config cfg, Wallet wallet) throws Exception {
+        // Possibly has to deploy contracts, so give it some extra time.
+        Context ctx = Prnm.contextWithTimeout(600);
+        try {
+            client = new Client(ctx, cfg, wallet);
+        } finally {
+            ctx.cancel();
+        }
+
+        // Start a new thread for handling channel proposals.
+        new Thread(() -> {
+            ProposalHandler propHandler = new ProposalHandler(this);
+            client.handleChannelProposals(propHandler);
+        }).start();
+    }
+
+    public void propose(Address peer, BigInts initBals, String ip, int port) throws Exception {
+        // This is safe to call more than once.
+        client.addPeer(peer, ip, port);
+        // Has to send transactions, so give it some extra time.
+        Context ctx = Prnm.contextWithTimeout(600);
+        try {
+            accept(client.proposeChannel(ctx, peer, 60, initBals));
+        } finally {
+            ctx.cancel();
+        }
+    }
+
+    @Override
+    public void accept(PaymentChannel channel) {
+        Log.i("prnm", "New channel " +channel.getIdx());
+        // Start a new thread for watching the channel.
+        new Thread(() -> {
+            try {
+                Log.d("channel", "Starting watching");
+                channel.watch();
+                Log.d("channel", "Stopped watching");
+            }  catch (Exception e) {
+                Log.e("channel", "Error watching:" + e.toString());
+            }
+        }).start();
+
+        // Start a new thread for handling channel updates.
+        new Thread(() -> {
+            UpdateHandler updateHandler = new UpdateHandler(channel);
+            Log.d("channel", "Starting handling updates");
+            channel.handleUpdates(updateHandler);
+            Log.d("channel", "Stopped handling updates");
+        }).start();
+    }
+}
+
+interface ChannelAcceptor {
+    // accept will be called on every new accepted channel.
+    public void accept(PaymentChannel channel);
+}
+
+class ProposalHandler implements prnm.ProposalHandler {
+    ChannelAcceptor acc;
+
+    public ProposalHandler(ChannelAcceptor acceptor) {
+        acc = acceptor;
+    }
+
+    // Handles all channel proposals by accepting them.
+    @Override
+    public void handle(ChannelProposal proposal, ProposalResponder responder) {
+        Context ctx = Prnm.contextWithTimeout(30);
```

Comment:
```suggestion
        Context ctx = Prnm.contextWithTimeout(600);
```
you need to cover a potential dispute settle in the handler context.

Reply:
Hm i fixed it in only two from three occurrences :sweat_smile: